src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,162 @@
+      {
+        basename <- basename(dataImportOptions$importLocation)
+        if (length(basename) > 0)
+        {
+          fileComponents <- unlist(strsplit(basename, ".", fixed = TRUE))
+          components <- length(fileComponents)
+          if (components >= 1)
+          {
+            dataName <- paste(fileComponents[1:components-1])
+          }
+        }
+      }
+    }
+  
+  dataName
+})
+
+.rs.addFunction("assemble_data_import_parameters_csv", function(dataImportOptions)
+{
+  parameters <- ""
+  spacing <- ",\n  "
+  useCsv <- (!is.null(dataImportOptions$delimiter) && dataImportOptions$delimiter == ",")
+
+  if (!is.null(dataImportOptions$delimiter) && dataImportOptions$delimiter != ",")
+  {
+    parameters <- paste0(parameters, spacing, "delim = \"", dataImportOptions$delimiter, "\"")
+  }
+
+  if (!is.null(dataImportOptions$quotes) && dataImportOptions$quotes != "" && !useCsv)
+  {
+    parameters <- paste0(parameters, spacing, "quote = \"", dataImportOptions$quotes, "\"")
+  }
+
+  if (!is.null(dataImportOptions$escapeBackslash) && !useCsv)
+  {
+    parameters <- paste0(parameters, spacing, "escape_backslash = ", dataImportOptions$escapeBackslash)
+  }
+
+  if (!is.null(dataImportOptions$escapeDouble) && !useCsv)
+  {
+    parameters <- paste0(parameters, spacing, "escape_double = ", dataImportOptions$escapeDouble)
+  }
+
+  if (!is.null(dataImportOptions$columnNames))
+  {
+    parameters <- paste0(parameters, spacing, "col_names = ", dataImportOptions$columnNames)
+  }
+
+  if (!is.null(dataImportOptions$trimSpaces) && useCsv)
+  {
+    parameters <- paste0(parameters, spacing, "trim_ws = ", dataImportOptions$trimSpaces)
+  }
+
+  if (!is.null(dataImportOptions$locale) && dataImportOptions$locale != "")
+  {
+    parameters <- paste0(parameters, spacing, "locale = readr::locale(date_names=\"", dataImportOptions$locale, "\")")
+  }
+
+  if (!is.null(dataImportOptions$na) && dataImportOptions$na != "")
+  {
+    parameters <- paste0(parameters, spacing, "na = \"", dataImportOptions$na, "\"")
+  }
+
+  if (!is.null(dataImportOptions$comments) && dataImportOptions$comments != "")
+  {
+    parameters <- paste0(parameters, spacing, "comments = \"", dataImportOptions$comments, "\"")
+  }
+
+  if (!is.null(dataImportOptions$skip))
+  {
+    parameters <- paste0(parameters, spacing, "skip = ", dataImportOptions$skip)
+  }
+
+  if (!is.null(dataImportOptions$maxRows))
+  {
+    parameters <- paste0(parameters, spacing, "n_max = ", dataImportOptions$maxRows)
+  }
+
+  parameters
+})
+
+.rs.addFunction("assemble_data_import", function(dataImportOptions)
+{
+  importInfo <- list()
+
+  dataName <- importInfo$dataName <- .rs.assemble_data_import_name(dataImportOptions)
+  if (is.null(dataName) || dataName == "")
+  {
+    dataName <- "dataset"
+  }
+
+  if (!grepl("^[a-zA-Z]+[a-zA-Z0-9_]*$", dataName) || any(dataName == ls('package:base')))
```

Comment:
I followed up from this comment and took a look at make.names(), wasn't super happy with the output, for instance, "my#$#data" becomes "my...data", would be nice to have something like "my_data" instead of replacing everything with periods. Therefore, my personal preference is to still keep the name as is, but I'm open to change this if more people chime in.